edgecast/resources/cps/certificate.go

```
@@ -0,0 +1,557 @@
+	}
+
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] certificate created: %# v", pretty.Formatter(resp))
+	log.Printf("[INFO] certificate id: %d", resp.ID)
+
+	d.SetId(strconv.FormatInt(resp.ID, 10))
+
+	return ResourceCertificateRead(ctx, d, m)
+}
+
+func ResourceCertificateRead(ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Not Yet Implemented
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ResourceCertificateUpdate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Not Yet Implemented
+	return ResourceCertificateRead(ctx, d, m)
+}
+
+func ResourceCertificateDelete(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Not Yet Implemented
+	var diags diag.Diagnostics
+	return diags
+}
+
+// expandDomains converts the Terraform representation of Domains into
+// the Domains API Model
+func expandDomains(attr interface{}) ([]*models.DomainCreateUpdate, error) {
+	if items, ok := attr.([]interface{}); ok {
+
+		domains := make([]*models.DomainCreateUpdate, 0)
+
+		for _, item := range items {
+			curr := item.(map[string]interface{})
+
+			domain := models.DomainCreateUpdate{
+				Name:         curr["name"].(string),
+				IsCommonName: curr["is_common_name"].(bool),
+			}
+
+			domains = append(domains, &domain)
+		}
+
+		return domains, nil
+
+	} else {
+		return nil,
+			errors.New("ExpandDomains: attr input was not a []interface{}")
+	}
+}
+
+// expandOrganization converts the Terraform representation of organization into
+// the Organization API Model
+func expandOrganization(attr interface{}) (*models.OrganizationDetail, error) {
+	if attr == nil {
+		return nil, nil
+	}
+
+	curr, err := helper.ConvertSingletonSetToMap(attr)
+	if err != nil {
+		return nil, fmt.Errorf("error expanding orgnization detail: %w", err)
+	}
+
+	// Empty map
+	if len(curr) == 0 {
+		return nil, nil
```

Would this be an error condition where we should return an error instead, or do we envision this in a go right flow?